### PR TITLE
Add dependency licenses check

### PR DESCRIPTION
**Pull request by kezhenxu94**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [x] New feature provided
- [ ] Improve performance

### CODE REVIEW

File: apm-dist/release-docs/NOTICE [hunk 1]
```diff
@@ -892,3 +892,12 @@ the 'license' directory of the distribution file, for the license terms of the
...
 ------
+
+========================================================================
+Freemarker NOTICE
+========================================================================
+Apache FreeMarker
+Copyright 2015-2018 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
```
Comment (wu-sheng) [inline apm-dist/release-docs/NOTICE]: Apache notice does not need to add. We are in the same foundation, read the first lines of this file.

File: oap-server/server-starter/src/main/resources/official_analysis.oal [hunk 2]
```diff
@@ -109,8 +109,8 @@ envoy_parent_connections_used = from(EnvoyInstanceMetric.value).filter(metricNam
...
-// disable(endpoint_relation_server_side);
-// disable(top_n_database_statement);
-// disable(zipkin_span);
-// disable(jaeger_span);
\ No newline at end of file
+disable(segment);
+disable(endpoint_relation_server_side);
+disable(top_n_database_statement);
+disable(zipkin_span);
+disable(jaeger_span);
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/official_analysis.oal]: Keep commented.

File: tools/dependencies/check-LICENSE.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# List all modules(jars) that belong to the SkyWalking itself, these will be ignored
+# when checking the dependency licenses
+./mvnw -Pbackend -Dexec.executable='echo' -Dexec.args='${project.artifactId}-${project.version}.jar' exec:exec -q > self-modules.txt
```
Comment (wu-sheng) [inline tools/dependencies/check-LICENSE.sh]: Why mvnw again?

File: tools/dependencies/check-LICENSE.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+ls dist/apache-skywalking-apm-bin/oap-libs > all-dependencies.txt
+
+# Exclude all self modules(jars) to generate all third-party dependencies
+grep -vf self-modules.txt all-dependencies.txt > third-party-dependencies.txt
+
+# Compare the third-party dependencies with known dependencies, expect that
+# all third-party dependencies are KNOWN and the exit code of the command is 0,
+# otherwise we should add its license to LICENSE file and add the dependency to known-dependencies.txt
+diff -U0 tools/dependencies/known-dependencies.txt third-party-dependencies.txt
```
Comment (wu-sheng) [inline tools/dependencies/check-LICENSE.sh]: You need to consider order. Maybe different in different OS.

Comment (kezhenxu94) [reply tools/dependencies/check-LICENSE.sh]: > You need to consider order. Maybe different in different OS.

I sort the dependencies when generating the files

File: tools/dependencies/check-LICENSE.sh (hunk 3 above)
Comment (kezhenxu94) [reply tools/dependencies/check-LICENSE.sh]: Not that we didn't package again here, we just list all the modules in the backend

**Pull request merged**